### ISSUE TITLE
NeruoNER python process crashes when using trained German model

### ISSUE THREAD

**Issue by ravikrsingh20** [Franck-Dernoncourt/NeuroNER#59]
I have trained an NER model for German language using NeuroNER.
Training details:
1. Created glove vector from 4GB German Wikipedia data. The size of the glove vector text file is 1.34GB
2. Gathered conll format German text for training from various sources online.
3. After training, a model of size 112MB is generated. The file size of dataset.pickle is 1.2GB

I tried to use the above generated model for tagging German texts but python processed crashed after some time. While executing NeuroNER, loading dataset is completed but after that the memory usage of python process increases to 6GB and then it crashed because of lack of RAM (8GB system).

Following is a snapshot of training results of epoch 3:

processed 5087519 tokens with 598708 phrases; found: 606123 phrases; correct: 509437.
accuracy:  96.50%; precision:  84.05%; recall:  85.09%; FB1:  84.57
              LOC: precision:  85.55%; recall:  90.46%; FB1:  87.93  274933
             MISC: precision:  71.78%; recall:  64.94%; FB1:  68.19  85075
              ORG: precision:  78.58%; recall:  75.29%; FB1:  76.90  87095
              PER: precision:  91.01%; recall:  94.12%; FB1:  92.54  159020

Evaluate model on the valid set
processed 1272248 tokens with 164754 phrases; found: 163204 phrases; correct: 136980.
accuracy:  96.13%; precision:  83.93%; recall:  83.14%; FB1:  83.54
              LOC: precision:  84.77%; recall:  88.42%; FB1:  86.56  76827
             MISC: precision:  75.32%; recall:  63.59%; FB1:  68.96  24165
              ORG: precision:  77.21%; recall:  73.02%; FB1:  75.06  22744
              PER: precision:  91.44%; recall:  93.91%; FB1:  92.66  39468

Evaluate model on the test set
processed 1587506 tokens with 209343 phrases; found: 207724 phrases; correct: 175447.
accuracy:  96.22%; precision:  84.46%; recall:  83.81%; FB1:  84.13
              LOC: precision:  85.21%; recall:  88.61%; FB1:  86.88  95109
             MISC: precision:  76.29%; recall:  65.70%; FB1:  70.60  31299
              ORG: precision:  77.83%; recall:  73.70%; FB1:  75.71  29458
              PER: precision:  91.78%; recall:  94.38%; FB1:  93.06  51858

Since during training also, the system must be tagging NER on test set using the generated model but this does not crash. I am not able to figure out why does this happen during executing NeuroNER on trained German model? Should I reduce glove vector dataset size or should i reduce training data?

Any help/guidance would be appreciated. 
Thanks a lot
